pkg/apis/hco/v1beta1/hyperconverged_types.go

```diff
@@ -56,25 +57,31 @@ type HyperConvergedConfig struct {
 	NodePlacement *sdkapi.NodePlacement `json:"nodePlacement,omitempty"`
 }
 
-type HyperConvergedFeatureGates map[string]bool
-
-func (fgs HyperConvergedFeatureGates) IsEnabled(fgName string) bool {
-	fg, found := fgs[fgName]
-	return fg && found
+// HyperConvergedFeatureGates is a set of optional feature gates to enable or disable new features that are not enabled
+// by default yet.
+// +optional
+// +k8s:openapi-gen=true
+type HyperConvergedFeatureGates struct {
+	// Allow attaching a data volume to a running VMI
+	// +optional
+	HotplugVolumes *bool `json:"HotplugVolumes,omitempty"`
 }
 
 // get list of feature gates from a specific operand list
-func (fgs HyperConvergedFeatureGates) GetFeatureGateList(candidates []string) []string {
-	res := make([]string, 0, len(fgs))
-	for _, fg := range candidates {
-		if fgs.IsEnabled(fg) {
-			res = append(res, fg)
-		}
+func (fgs *HyperConvergedFeatureGates) GetFeatureGateList() []string {
+	res := make([]string, 0, 1)
+
+	if fgs.IsHotplugVolumesEnabled() {
+		res = append(res, virtconfig.HotplugVolumesGate)
 	}
 
 	return res
 }
 
+func (fgs HyperConvergedFeatureGates) IsHotplugVolumesEnabled() bool {
```

Comment:
Here too, it may be more usable to have a pointer receiver (since `Spec.FeatureGates` could be `nil`), and support the case in which `fgs == nil`. This will make the calls a bit simpler, without needing to check if `Spec.FeaturesGates != nil`.